vault/resource_mfa_duo_test.go

```diff
@@ -0,0 +1,78 @@
+package vault
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"os"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestMFADuoBasic(t *testing.T) {
+
+	isEnterprise := os.Getenv("TF_ACC_ENTERPRISE")
+	if isEnterprise == "" {
+		t.Skip("TF_ACC_ENTERPRISE is not set, test is applicable only for Enterprise version of Vault")
+	}
+
+	mfaDuoPath := acctest.RandomWithPrefix("mfa-duo")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testMFADuoConfig(mfaDuoPath),
+				Check:  testMFADuoCheckAttrs(),
+			},
+			{
+				Config:      testMFADuoConfig(mfaDuoPath + "/"),
+				Destroy:     false,
+				ExpectError: regexp.MustCompile("vault_mfa_duo\\.test: cannot write to a path ending in '/'"),
+			},
+		},
+	})
+}
+
+func testMFADuoCheckAttrs() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resourceState := s.Modules[0].Resources["vault_mfa_duo.test"]
+		if resourceState == nil {
+			return fmt.Errorf("resource not found in state")
+		}
+
+		instanceState := resourceState.Primary
+		if instanceState == nil {
+			return fmt.Errorf("resource has no primary instance")
+		}
+
+		return nil
+	}
+}
+
+func testMFADuoConfig(path string) string {
+
+	userPassPath := acctest.RandomWithPrefix("userpass")
+
+	return fmt.Sprintf(`
+resource "vault_auth_backend" "userpass" {
+  type = "userpass"
+  path = %q
+}
+
+resource "vault_mfa_duo" "test" {
+  name                  = %q
+  mount_accessor        = "${vault_auth_backend.userpass.accessor}"
+  secret_key            = "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz"
+  integration_key       = "BIACEUEAXI20BNWTEYXT"
```

Comment:
I assume these are fake but just wanted to point them out in case.

Reply:
Yep, I took these straight from the API docs: https://www.vaultproject.io/api/system/mfa/duo.html#sample-payload

Reply:
:+1: that's what I thought! Thanks for confirming!